packages/x-charts/src/LineChart/AnimatedLine.tsx

```diff
@@ -16,6 +17,32 @@ export interface AnimatedLineProps extends React.ComponentPropsWithoutRef<'path'
   skipAnimation?: boolean;
 }
 
+type LineAnimatedProps = Pick<AnimatedLineProps, 'd' | 'skipAnimation'> & {
+  ref?: React.Ref<SVGPathElement>;
+};
+
+/** Animates a line of a line chart using a `path` element.
+ * The props object also accepts a `ref` which will be merged with the ref returned from this hook. This means you can
+ * pass the ref returned by this hook to the `path` element and the `ref` provided as argument will also be called. */
+export function useAnimateLine(props: LineAnimatedProps): {
+  ref: React.Ref<SVGPathElement>;
+  d: string;
+} {
+  const ref = useAnimate(
+    { d: props.d },
+    {
+      createInterpolator: (lastProps, newProps) => {
+        const interpolate = interpolateString(lastProps.d, newProps.d);
+        return (t) => ({ d: interpolate(t) });
+      },
+      applyProps: (element: SVGPathElement, { d }) => element.setAttribute('d', d),
+      skip: props.skipAnimation,
+    },
+  );
+
+  return { ref: useForkRef(ref, props.ref), d: props.d };
+}
```

Comment:
`UseAnimatedLineParams` seems better suited than `LineAnimatedProps` in this case. Also added the return value type.

```suggestion
type UseAnimatedLineParams = Pick<AnimatedLineProps, 'd' | 'skipAnimation'> & {
  ref?: React.Ref<SVGPathElement>;
};

type UseAnimatedLineReturnValue = {
  ref: React.Ref<SVGPathElement>;
  d: string;
}

/** Animates a line of a line chart using a `path` element.
 * The props object also accepts a `ref` which will be merged with the ref returned from this hook. This means you can
 * pass the ref returned by this hook to the `path` element and the `ref` provided as argument will also be called. */
export function useAnimateLine(params: UseAnimatedLineParams): UseAnimatedLineReturnValue {
  const ref = useAnimate(
    { d: params.d },
    {
      createInterpolator: (lastProps, newProps) => {
        const interpolate = interpolateString(lastProps.d, newProps.d);
        return (t) => ({ d: interpolate(t) });
      },
      applyProps: (element: SVGPathElement, { d }) => element.setAttribute('d', d),
      skip: params.skipAnimation,
    },
  );

  return { ref: useForkRef(ref, params.ref), d: params.d };
}
```